### DISCUSSION TITLE
JPA Blaze criteria with entityview - selective output

### DISCUSSION THREAD

**Discussion by adamzrk** [Blazebit/blaze-persistence#1962] [Q&A]
I am using blaze persistence framework with entityviews. Suppose I have enetity view like

```
@EntityView
interface User {
   String getName();
   String getMobile();
   List<Address> getAddresses()
}
```
And, based on search criteria, I would like to have all the data in the results or some part of data data - i.e. user without addresses if those are not to be displayed in the output. Is there a way to achieve that in the code? I am interesed in blaze persistence mostly but standard jpa/hibernate is also an option.

**Comment by beikov** [comment]
Yes, you can fetch a data subset. Also see https://persistence.blazebit.com/documentation/1.6/entity-view/manual/en_US/index.html#fetching-a-data-subset

**Reply by adamzrk** [reply]
Thank you for reply, but is there possibility to fetch collections like in the documentation?
I have problem when doing:
`entityViewSetting.fetch("adresses");`

which is mapped in my main entity view as
``` 
@Mapping(value = "adresses", fetch = FetchStrategy.MULTISET)
Set<AdresseView> getAdresses();
```

when I get the resultList from criteria then exception is thrown:

>Caused by: java.lang.ClassCastException: class java.util.HashSet cannot be cast to class java.lang.String (java.util.HashSet and java.lang.String are in module java.base of loader 'bootstrap')

**Reply by beikov** [reply]
Looks like a bug. It should work with the JOIN strategy though.

**Reply by adamzrk** [reply]
Thabk you bitllut thats not good as I need multiset for relations/collections and I cannot achieve that using join.

**Reply by beikov** [reply]
Please create a reproducer application that shows the issue e.g. based on [one of our quickstarts](https://github.com/Blazebit/blaze-persistence?tab=readme-ov-file#quickstart) and attach that to a new issue on GitHub.
